internal/pkg/consensus/election.go

```diff
@@ -46,26 +48,26 @@ func (em ElectionMachine) GenerateElectionProof(ctx context.Context, entry *dran
 }
 
 // GenerateWinningPoSt creates a PoSt proof over the input miner ID and sector infos.
-func (em ElectionMachine) GenerateWinningPoSt(allSectorInfos []abi.SectorInfo, entry *drand.Entry, epoch abi.ChainEpoch, ep postgenerator.PoStGenerator, maddr address.Address) ([]abi.PoStProof, error) {
-	minerID, err := address.IDFromAddress(maddr)
+func (em ElectionMachine) GenerateWinningPoSt(ctx context.Context, allSectorInfos []abi.SectorInfo, entry *drand.Entry, epoch abi.ChainEpoch, ep postgenerator.PoStGenerator, maddr address.Address) ([]abi.PoStProof, error) {
+	minerIDuint64, err := address.IDFromAddress(maddr)
 	if err != nil {
 		return nil, err
 	}
-	_ = minerID
+	minerID := abi.ActorID(minerIDuint64)
 
 	seed := blake2b.Sum256(entry.Signature)
 	randomness, err := crypto.BlendEntropy(acrypto.DomainSeparationTag_ElectionPoStChallengeSeed, seed[:], epoch, []byte{})
 	if err != nil {
 		return nil, err
 	}
-	_ = randomness
+	poStRandomness := abi.PoStRandomness(randomness)
 
-	// challengeIndexes, err := ep.GenerateWinningPoStSectorChallenge(proofs[0].RegisteredProof, minerID, randomness, len(allPrivateSectorInfos))
-	// if err != nil {
-	// 	return nil, err
-	// }
-	// challengedSectorInfos := filterSectorInfosByIndex(allSectorInfos, challengeIndexes)
-	// return ep.GenerateWinningPoSt(minerID, challengedSectorInfos, randomness)
+	challengeIndexes, err := ffiwrapper.ProofVerifier.GenerateWinningPoStSectorChallenge(ctx, allSectorInfos[0].RegisteredProof, minerID, poStRandomness, uint64(len(allSectorInfos)))
```

Comment:
This is more complicated than it sounds. The verification below is actually getting an EPoStVerifier that has that method. the EPoStVerifier interface that this method uses doesn't have the generate challenges method, and it's hard to add because it's actually fulfilled by a `Prover` interface that comes out of sector stroage. Sadly all of these interfaces are just ffiwrapper under the hood.